stix/common/kill_chains.py

```diff
@@ -204,7 +245,102 @@ class KillChainPhasesReference(stix.EntityList):
     _binding_var = "Kill_Chain_Phase"
     _inner_name = "kill_chain_phases"
 
+    def _is_lmco(self, value):
+        return any(value.phase_id == x.phase_id for x in LMCO_KILL_CHAIN_PHASES)
+
+    def _fix_value(self, value):
+        if isinstance(value, KillChainPhase) and self._is_lmco(value):
```

Comment:
I guess that `@kill_chain_id` isn't required, so we could possibly expand the type promotion beyond LMCO phases, omitting `@kill_chain_id` if it isn't an LMCO phase.